### DISCUSSION TITLE
Grammarly for Mac is unstable

### DISCUSSION THREAD

**Discussion by anaclumos** [notaapp/talk#317] [Questions] [answered]
Grammarly for Mac doesn't work with Nota. If I click on the suggestion, instead of Grammarly replacing the wrong word, Grammarly appends the suggested word to the end of the sentence.
Given that Grammarly for Mac works fine on every other Mac Apps with `UITextField` or `NSTextField`, this seems like a Nota issue.

**Comment by astoilkov** [comment]
Yes, it is. I will look into this.

Thanks for this and all other issues!

**Comment by astoilkov** [comment, marked as answer]
I've spent quite a bit of time researching this. Here is what I learned. 

Grammarly detects native OS components in order to provide spellchecking. However, we don't use a standard OS component for displaying text. We use a hidden text component that we use to detect typing. This makes it impossible to make Grammarly fully compatible with our current implementation.

I've tested other apps as well. Those who use native OS components work, those who don't have issues.

I will continue my research on this. I see two possible solutions in the future:
- Grammarly to add an API that you can send text and receive errors.
- We can fundamentally change how our editor works and support this case. This is something we are considering because it has other benefits as well.

**Reply by davidaayers** [reply]
I know this is an old question/issue, but it would be great to have a way to use Grammarly with Nota. Grammarly is a big part of my workflow, such that muscle memory kicks in and ends up messing up the text in my notes :)
